test/e2e/test_worker_status.py

```diff
@@ -40,3 +42,83 @@ def test_worker_lifecycle_status_is_expected(
             fleet_id=deadline_resources.fleet.id,
             worker_id=function_worker.worker_id,
         )
+
+    def test_linux_worker_restarts_process(
+        self,
+        deadline_resources,
+        deadline_client: DeadlineClient,
+        function_worker: EC2InstanceWorker,
+    ) -> None:
+        # Verifies that Linux Worker service restarts the process when we start/stop worker process
+
+        assert function_worker.worker_id is not None  # To fix linter type mismatch
+
+        assert is_worker_started(
+            deadline_client=deadline_client,
+            farm_id=deadline_resources.farm.id,
+            fleet_id=deadline_resources.fleet.id,
+            worker_id=function_worker.worker_id,
+        )
+
+        # First check that the worker service is running
+        service_check_result = function_worker.send_command("systemctl is-active deadline-worker")
+        assert (
+            service_check_result.exit_code == 0
+        ), "Unable to check whether deadline-worker is active"
+        assert (
+            "inactive" not in service_check_result.stdout
+            and "active" in service_check_result.stdout
+        ), f"deadline-worker is in unexpected status {service_check_result.stdout}"
+
+        # Check that the worker process is running
+
+        process_check_result = function_worker.send_command(
+            f"pgrep --count --full -u {function_worker.configuration.agent_user} deadline-worker-agent"
+        )
+
+        assert process_check_result.exit_code == 0, "deadline-worker-agent process is not running"
+
+        time_that_worker_was_killed: datetime = datetime.now(timezone.utc)
+        pkill_command_result = function_worker.send_command(
+            f"sudo pkill -9 --full -u {function_worker.configuration.agent_user} deadline-worker-agent"
+        )
+        assert (
+            pkill_command_result.exit_code == 0
+        ), f"Failed to kill the worker agent process: {pkill_command_result}"
+
+        # Wait for the process to be restarted by the service
+
+        time.sleep(10)
```

Comment:
nit: I'm pretty sure 10 seconds is more than enough for this test, but I think a better pattern would be to build in retries with a timeout (10 seconds is plenty, you can probably check after a second or two even) so we at least don't give up on the first failure for this check. Magic sleeps like this tend to lead to flakiness in the tests.

Reply:
That's true, I'll create something like that